generic/tests/dd_test.py

```diff
@@ -51,73 +52,101 @@ def _get_file(filename, select):
                 err = ("Incorrect cfg: dd_select out of the range (disks=%s,"
                        " select=%s)" % (disks, select))
                 logging.error(err)
-                raise error.TestError(err)
+                test.error(err)
+
+    class CmdFormat(object):
+        def __init__(self, params):
+            for key in dd_keys:
+                setattr(self, key, params.get(key))
+            self.dd_if_select = int(params.get("dd_if_select", '-1'))
+            self.dd_of_select = int(params.get("dd_of_select", '-1'))
+            self.partitioned_device = []
+
+        def _add_suffix(self, dd_cmd_list, suffix):
+            if isinstance(suffix, six.string_types):
+                for i in range(len(dd_cmd_list)):
+                    dd_cmd_list[i] += suffix
+            else:
+                assert len(dd_cmd_list) == len(suffix), "lengths mismatch"
+                for i in range(len(suffix)):
+                    dd_cmd_list[i] += str(suffix[i])
+
+        def _get_cmd(self):
+            dd_bs_list = self.dd_bs.split() if self.dd_bs else []
```

Comment:
Since there are many cases `params.get(bs)` would be `None`, this may be more efficient. 
```python
dd_bs_list, dd_cmd_list = ([], ['dd'])
if self.dd_bs:
    dd_bs_list = [' bs=' + bs for bs in self.dd_bs.split()]
    dd_cmd_list *= len(dd_bs_list)
```
It's up to you.